### ISSUE TITLE
Epoch Loss not getting updated properly?

### ISSUE THREAD

**Issue by waydegilliam** [damitkwr/ESRNN-GPU#21]
Edit: Never mind. The loss is just getting averaged as `batch_num` is outside of the for-loop that it increases in.

In `trainer.py` inside the `train` method after an epoch finishes, the `epoch_loss` is divided by the `batch_num` + 1. That means that after every batch, the `epoch_loss` is forcefully decreased as the denominatior (the `batch_num`) is constantly getting bigger:

`epoch_loss = epoch_loss / (batch_num + 1)`

Maybe I'm misunderstanding something here, but it doesn't seem right that the loss is getting artificially decreased simply based on which batch the training loop is on. I looked through the original C++ implementation, but couldn't find anything that looked like the above line (I don't know C++ very well, so that may be why).

P.S.
Thanks for the python/torch implementation of this project btw, it's a great resource for learning some good forecasting methods/strategies.
